src/com/superliminal/magiccube4d/PolytopePuzzleDescription.java

```diff
@@ -266,15 +268,50 @@ private static void Assumpt(boolean condition) {
      * omnitruncated regular
      */
 
-    public PolytopePuzzleDescription(String schlafliProduct,
-        double length, // usually int but can experiment with different cut depths
-        ProgressCallbacks progress)
+    public PolytopePuzzleDescription(String prescription, ProgressCallbacks progress)
     {
-        this.schlafliProduct = schlafliProduct;
-        this.edgeLength = length;
+        // Caller may have passed us "Grand Antiprism" from a menu,
+        // but the lower level function we're going go call doesn't allow spaces,
+        // and understands Grand_Antiprism as a special case, so convert.
+        prescription = prescription.replaceAll("Grand Antiprism",
+                                               "Grand_Antiprism");
+        java.util.regex.Matcher matcher =
+        java.util.regex.Pattern.compile(
+            "\\s*([^ ]+)\\s+((\\d+)(\\((.*)\\))?)"
+        ).matcher(prescription);
+        if (!matcher.matches())
+            throw new IllegalArgumentException("PolytopePuzzleDescription didn't understand prescription string "+com.donhatchsw.util.Arrays.toStringCompact(prescription)+"");
+	String schlafliProductString = matcher.group(1);
+        String lengthString = matcher.group(2);
+        String intLengthString = matcher.group(3);
+        String doubleLengthString = matcher.group(5);
+
+        int intLength = Integer.parseInt(intLengthString);
+        double doubleLength = (double)intLength; // but to be overridden by doubleLengthString if there is one
+        if (doubleLengthString != null)
+        {
+            // Allow fractions
+            int slashIndex = doubleLengthString.lastIndexOf('/');
+            if (slashIndex != -1)
+            {
+                String numeratorString = doubleLengthString.substring(0, slashIndex);
+                String denominatorString = doubleLengthString.substring(slashIndex+1);
+                doubleLength = Double.valueOf(numeratorString).doubleValue()
+                             / Double.valueOf(denominatorString).doubleValue();
+            }
+            else
+                doubleLength = Double.valueOf(doubleLengthString).doubleValue(); //  XXX should catch parse error and throw illegal arg exception
+        }
+
+        this.schlafliProduct = schlafliProductString;
+        this.lengthString = lengthString;
+        this.intLength = intLength;
+        this.doubleLength = doubleLength;
 
-        if(length < 1)
-            throw new IllegalArgumentException("PolytopePuzzleDescription called with length=" + length + ", min legal length is 1");
+        if (intLength < 1)
+            throw new IllegalArgumentException("PolytopePuzzleDescription called with intLength="+intLength+", min legal intLength is 1");
+        if (doubleLength <= 0)
```

Comment:
Since it is 1/doubleLength that ends up used, does it make sense to allow values <= 1?

Reply:
Yes!
E.g. center cut dodecahedron is "{5,3} 2(0.7639320225002102)"  (where .7639... = 3-sqrt(5)):
![center_cut_dodecahedron](https://user-images.githubusercontent.com/12412901/92324748-5a5a8b80-eff9-11ea-83c2-06d48b7d32c6.png)

Reply:
ah, of course, thanks!